pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslice
+
+import (
+	"context"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/discovery"
+	"k8s.io/kubernetes/pkg/apis/discovery/validation"
+)
+
+// endpointSliceStrategy implements verification logic for Replication.
+type endpointSliceStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating Replication EndpointSlice objects.
+var Strategy = endpointSliceStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns true because all EndpointSlices need to be within a namespace.
+func (endpointSliceStrategy) NamespaceScoped() bool {
+	return true
+}
+
+// PrepareForCreate clears the status of an EndpointSlice before creation.
+func (endpointSliceStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	endpointSlice := obj.(*discovery.EndpointSlice)
+	endpointSlice.Generation = 1
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (endpointSliceStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newEndpointSlice := obj.(*discovery.EndpointSlice)
```

Comment:
FWIW go conventions prefer shorter names for local vars.  oldEPS newEPS would be OK :)